auto_nag/bzcleaner.py

```diff
@@ -219,6 +234,82 @@ def add_auto_ni(self, bugid, data):
             }
         return True
 
+    def add_prioritized_action(self, bug, quota_name, needinfo=None, autofix=None):
+        """
+        - `quota_name` is the key used to apply the limits, e.g., triage owner, team, or component
+        """
+        assert needinfo or autofix
+
+        # Avoid having more than one ni from our bot
+        if needinfo and self.has_bot_set_ni(bug):
+            needinfo = autofix = None
+
+        action = {
+            "bugid": str(bug["id"]),
+            "needinfo": needinfo,
+            "autofix": autofix,
+        }
+
+        sort_key = self.get_bug_sort_key(bug)
+        if sort_key is not None:
+            action["key"] = sort_key
+
+        self.quota_actions[quota_name].append(action)
+
+    def get_bug_sort_key(self, bug):
+        return None
+
+    def _populate_prioritized_actions(self, bugs):
+        max_actions = self.get_max_actions()
+        max_ni = self.get_max_ni()
+        exclude_no_action_bugs = (
+            len(self.quota_actions) > 0 and self.exclude_no_action_bugs()
+        )
+        bugs_with_action = set() if exclude_no_action_bugs else dummySet
```

Comment:
Did you use the dummySet for optimization purposes?
If so, I feel this is a case of premature optimization and I'd prefer not sacrificing clarity for a likely minor optimization

Reply:
Yes, it is just an optimization. I agree with you; now I look at it again, I see it may introduce unnecessary complexity. Simplified in 45080abc59808c7a502141a99fad2c4d5d69486a.